sway-parse/src/parser.rs

```diff
@@ -153,6 +223,43 @@ impl<'a, 'e> Parser<'a, 'e> {
     pub fn full_span(&self) -> &Span {
         &self.full_span
     }
+    /// Consume tokens while its line equals to `line`.
+    ///
+    /// # Warning
+    ///
+    /// To calculate lines the original source code needs to be transversed.
+    pub fn consume_while_line_equals(&mut self, line: usize) {
+        loop {
+            let Some(current_token) = self.token_trees.get(0) else {
+                break;
+            };
+
+            let current_span = current_token.span();
+            let current_span_line = current_span.start_pos().line_col().0;
+
+            if current_span_line != line {
+                break;
+            } else {
+                self.token_trees = &self.token_trees[1..];
+            }
+        }
+    }
+
+    pub fn clear_errors(&mut self) {
+        self.handler.clear_errors();
+    }
+
+    pub fn clear_warnings(&mut self) {
+        self.handler.clear_errors();
+    }
```

Comment:
I'm very much against the idea of clearing all errors in the handlers like this because it introduces an easy way to remove meaningful errors that are outside of one's scope.

If you want to conditionally append errors, you should pass a new empty Handler and append it to the global one or not depending on whether you want to commit to the errors. Mutating the existing error set I don't think is acceptable.